src/bno055_uart.cpp

```diff
@@ -59,57 +60,60 @@ bool BNO055UART::isOpen()
 void BNO055UART::read(uint8_t page)
 {
   if (page != page_ && read_command_.address != BNO055Register::PAGE_ID) {
-    // TODO: Change pages automtically
+    // TODO(cottsay): Change pages automtically
     throw serial::SerialException("Page change required");
   }
 
   for (size_t this_write, index = 0; index < sizeof(read_command_); index += this_write) {
-    this_write = port_.write((const uint8_t *)&read_command_ + index, 1);
+    this_write = port_.write(reinterpret_cast<const uint8_t *>(&read_command_) + index, 1);
```

Comment:
I'm getting a major code smell here, and it's hard to tell what you're doing. If you must do it this way,
1. move the reinterpret cast outside the loop.
2. get rid of `this_write`. If it's zero, you're throwing an exception and you're passing 1 for the length.

Better yet, use port_.write(reinterpret_cast<const uint8_t *>(&read_command_),sizeof(read_command_))

Reply:
I refactored the `port_` reference, offsetting and cast into a separate function. It cleaned up the code pretty substantially.